kernel/kernel-api/src/main/java/io/delta/kernel/Scan.java

```diff
@@ -106,6 +110,16 @@ public interface Scan {
      */
     Row getScanState(TableClient tableClient);
 
+    /**
+     * Get the physical data read schema to pass to the parquet handler to read the given data file.
+     *
+     * @param tableClient instance of {@link TableClient} to use.
+     * @param scanState   Scan state {@link Row}
+     * @param scanFile   Scan file {@link Row}
+     * @return Physical schema to read from the data files.
+     */
+    StructType getPhysicalDataReadSchema(TableClient tableClient, Row scanState, Row scanFile);
```

Comment:
Should this be a static method? This method gets called on executors where there is no `Scan` object.